src/Build/Evaluation/Expander.cs

```diff
@@ -1582,6 +1578,16 @@ private static object ExpandMSBuildThisFileProperty(string propertyName, IElemen
             /// </summary>
             private static string ExpandRegistryValue(string registryExpression, IElementLocation elementLocation)
             {
+#if RUNTIME_TYPE_NETCORE
+                // .NET Core MSBuild used to always return empty, so match that behavior
+                // on non-Windows (no registry), and with a changewave (in case someone
+                // had a registry property and it breaks when it lights up).
+                if (!NativeMethodsShared.IsWindows || !ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_4))
```

Comment:
Do you need the ifdef here? People are opted in anyway, so it won't affect anyone unless they disable the change wave, and I don't think people normally do that. Or maybe just plan to remove the ifdef with the change wave? If so, that should be in a comment so we remember.

Reply:
Same for other cases.